habitat/datasets/rearrange/samplers.py

```diff
@@ -615,18 +687,29 @@ def sample(
                 # NOTE: only query and set pose once per instance for efficiency
                 pose = ao_instance.joint_positions
                 for link_ix, joint_range in link_ranges.items():
-                    if (
-                        ao_instance.handle == receptacle.parent_object_handle
-                    ) and (
-                        (link_ix == receptacle.parent_link)
-                        or ("frige" in receptacle.parent_object_handle)
-                        or ("fridge" in receptacle.parent_object_handle)
-                    ):
+                    should_sample_all_joints = joint_range[2]
+                    if receptacle is not None:
+                        if (
+                            ao_instance.handle
+                            == receptacle.parent_object_handle
+                        ) and (
+                            (link_ix == receptacle.parent_link)
+                            or should_sample_all_joints
+                        ):
+                            # If this is true, this means that the receptacle AO must be opened. That is because
+                            # the object is spawned inside the fridge OR inside the kitchen counter BUT not on top of the counter
+                            # because in this case all drawers must be closed.
```

Comment:
```suggestion
                            # because in this case all drawers must be closed.
                            # TODO: move this receptacle access logic to the ao_config files in a future refactor
```